pkg/controllers/jobset_controller.go

```diff
@@ -314,6 +316,48 @@ func updateReplicatedJobsStatuses(ctx context.Context, js *jobset.JobSet, status
 	updateStatusOpts.shouldUpdate = true
 }
 
+func updateJobSetStatus(_ context.Context, js *jobset.JobSet, updateStatusOpts *statusUpdateOpts) {
+	var runningJobs int32
+	var failedJobs int32
+	var completedJobs int32
+	var suspendedJobs int32
+	// Count the number of active, failed, completed, and suspended jobs.
+	for _, replicatedJobStatus := range js.Status.ReplicatedJobsStatus {
+		runningJobs += replicatedJobStatus.Active
+		failedJobs += replicatedJobStatus.Failed
+		completedJobs += replicatedJobStatus.Succeeded
+		suspendedJobs += replicatedJobStatus.Suspended
+	}
+
+	status := jobset.Pending
+	// If all jobs are completed, the status is "successful".
+	// If one job are failed, the status is "failed".
+	// If one job is suspended, the status is "suspended".
+	// Otherwise, the status is "running".
```

Comment:
We already have the conditions set, `Phase` should simply be a reflection of that. So the `Phase` should be set in the following place based on the condition  https://github.com/kubernetes-sigs/jobset/blob/390fac27fad21cc79c2f2e7af9e64b53c10f710b/pkg/controllers/jobset_controller.go#L875

Reply:
thanks for reviewing!
I roughly understand the meaning, so I will use the following method to modify it.
like this
```go
// function parameters for setCondition
type conditionOpts struct {
	eventType string
        // new add phase field
        phase.        string
	condition *metav1.Condition
}

```

Reply:
I might need to think about this again. sorry